lxd-agent/network.go

```diff
@@ -69,3 +78,101 @@ func serverTLSConfig() (*tls.Config, error) {
 
 	return tlsConfig, nil
 }
+
+// reconfigureNetworkInterfaces checks for the existence of "nics.json" in the config share.
+// Interfaces are identified by their MAC address and configured accordingly.
+func reconfigureNetworkInterfaces() {
+	// Abort if configuration file does not exist (nothing to do), otherwise log and return.
+	nicDir, err := ioutil.ReadDir("nics")
+	if os.IsNotExist(err) {
+		return
+	} else if err != nil {
+		logger.Errorf("Could not read network interface configuration directory: %s", err.Error())
+		return
+	}
+
+	nicData := make(map[string]deviceConfig.NICConfig)
+	for _, f := range nicDir {
+		nicBytes, err := ioutil.ReadFile(filepath.Join("nics", f.Name()))
+		if err != nil {
+			logger.Errorf("Could not read network interface configuration file: %s", err.Error())
+		}
+
+		var conf deviceConfig.NICConfig
+		err = json.Unmarshal(nicBytes, &conf)
+		if err != nil {
+			logger.Errorf("Could not parse network interface configuration file: %s", err.Error())
+			return
+		}
+
+		nicData[f.Name()] = conf
+	}
+
+	netDir := "/sys/class/net"
+	files, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return
+	}
+
+	reverter := revert.New()
+	for _, f := range files {
+		addrBytes, err := ioutil.ReadFile(filepath.Join(netDir, f.Name(), "address"))
+		if err != nil {
+			return
+		}
+		macAddress := string(bytes.TrimRight(addrBytes, "\n"))
```

Comment:
Probably need to parse this into `net.HardwareAddress` see https://github.com/lxc/lxd/pull/9893/commits/b944e46935de4fc832585709c321a873a11b6506#r804736771